Userland/Libraries/LibCore/SecretString.cpp

```diff
@@ -14,7 +18,11 @@ SecretString SecretString::take_ownership(char*& cstring, size_t length)
     auto buffer = ByteBuffer::copy(cstring, length);
     VERIFY(buffer.has_value());
 
+#if defined(AK_OS_MACOS)
```

Comment:
Do you think it makes sense to have a local  static function that wraps this logic, so it can be shared with both callers?

Reply:
It's only twice, so I don't think it's particularly necessary, but if you think we should do it then I'll make that change. Would like to get this merged soon though since I was working on something else actually my compilation broke :^(